Mastodon/Scene/Profile/UserLIst/UserListViewModel+State.swift

```diff
@@ -123,40 +124,61 @@ extension UserListViewModel.State {
             
             Task {
                 do {
-                    let response: Mastodon.Response.Content<[Mastodon.Entity.Account]>
+                    let accountResponse: Mastodon.Response.Content<[Mastodon.Entity.Account]>
                     switch viewModel.kind {
                     case .favoritedBy(let status):
-                        response = try await viewModel.context.apiService.favoritedBy(
+                            accountResponse = try await viewModel.context.apiService.favoritedBy(
                             status: status,
                             query: .init(maxID: maxID, limit: nil),
                             authenticationBox: viewModel.authContext.mastodonAuthenticationBox
                         )
                     case .rebloggedBy(let status):
-                        response = try await viewModel.context.apiService.rebloggedBy(
+                            accountResponse = try await viewModel.context.apiService.rebloggedBy(
                             status: status,
                             query: .init(maxID: maxID, limit: nil),
                             authenticationBox: viewModel.authContext.mastodonAuthenticationBox
                         )
                     }
 
+                    if accountResponse.value.isEmpty {
+                        await enter(state: NoMore.self)
+
+                        viewModel.accounts = []
+                        viewModel.relationships = []
+                        return
+                    }
+
                     var hasNewAppend = false
-                    var userIDs = viewModel.userFetchedResultsController.userIDs
-                    for user in response.value {
-                        guard !userIDs.contains(user.id) else { continue }
-                        userIDs.append(user.id)
+
+                    let newRelationships = try await viewModel.context.apiService.relationship(forAccounts: accountResponse.value, authenticationBox: viewModel.authContext.mastodonAuthenticationBox)
+
+                    var accounts = viewModel.accounts
```

Comment:
I _think_ this is not doing what you want. You want a copy of the accounts right? This way you'll likely get a reference to the wrapped property.

I've done the same before in the `StatusDataController` (new Name 🎉 ) and had to instantiate an array using the contents of the wrapped property to get a copy: https://github.com/mastodon/mastodon-ios/pull/1167/files#diff-ebefe3e24206519ef132e8d197c0b75529b20348df9de90d6e1465c0f136131eR39

Reply:
That's actually a very interesting point, so I had to dig into this 🤓 

As far as I'm aware, structs are value types and as `Array` is a struct, `Array` is also a value type. I also checked the [documentation on swift.org](https://www.swift.org/documentation/articles/value-and-reference-types.html) and they say:

> In Swift, structures, enumerations, and tuples are all value types.

But what about `accounts` as they're `@Published`? Well:

```
(lldb) po type(of: viewModel.accounts)
Swift.Array<MastodonSDK.Mastodon.Entity.Account>
```

Well, to be honest, I debugged this with a breakpoint in line 159 first 🫣 

```
// before line 159
(lldb) po viewModel.accounts
0 elements
(lldb) po accounts
0 elements
// after line 159
(lldb) po viewModel.accounts
0 elements
(lldb) po accounts
▿ 1 element
  ▿ 0 : <Account: 0x115006640>
```

So to me it looks like it works as expected. But maybe I am missing something here?